pkg/gds/admin/v2/csurf_test.go

```diff
@@ -0,0 +1,175 @@
+package admin_test
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/cookiejar"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/require"
+	"github.com/trisacrypto/directory/pkg/gds/admin/v2"
+)
+
+func TestDoubleCookies(t *testing.T) {
+	// Test both the DoubleCookie middleware and the SetDoubleCookieTokens handler.
+	router := gin.New()
+
+	// Add login route that sets the cookies
+	router.GET("/login", func(c *gin.Context) {
+		err := admin.SetDoubleCookieTokens(c, time.Now().Add(time.Minute*10).Unix())
+		require.NoError(t, err)
+		c.JSON(http.StatusOK, gin.H{"success": true})
+	})
+
+	// Add request route that requires double cookie submit
+	router.POST("/action", admin.DoubleCookie(), func(c *gin.Context) {
+		c.JSON(http.StatusCreated, gin.H{"success": true})
+	})
+
+	// Create an http client with a cookie jar
+	jar, err := cookiejar.New(nil)
+	require.NoError(t, err)
+	client := http.Client{
+		Jar: jar,
+	}
+
+	// Create the test server with the CSRF protected router
+	server := httptest.NewServer(router)
+
+	// Attempt to make a request to the server that is not CSRF protected
+	req, err := http.NewRequest(http.MethodPost, server.URL+"/action", nil)
+	require.NoError(t, err)
+
+	// Ensure the request is Forbidden
```

Comment:
Is it customary to say the request is Forbidden rather than that the response is Forbidden? Just trying to get the hang of front end syntax

Reply:
I suppose my comment is somewhat non-standard. I mean to say that the response from the server to this request has a 403 Forbidden status code and status.